### ISSUE TITLE
TimeoutError: waiting for selector "[aria-label="Listen only"]" failed: timeout 60000ms exceeded

### ISSUE THREAD

**Issue by awe007** [jibon57/bbb-recorder#22]
Good day!
I have error, when i try to use live broadcast function.
BigBlueButton and bbb-recorder are on the same local server.

In one window of the terminal, I run the command
`node ffmpegServer.js`

In second terminal screen i enter this command:
`node liveRTMP.js "https://domain.online/bigbluebutton/api/join?meetingId=adm-96w-m2a`

and after some seconds i see this error
```
PAGE LOG: FFMPEG_SERVER
{ TimeoutError: waiting for selector "[aria-label="Listen only"]" failed: timeout 60000ms exceeded
    at new WaitTask (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:549:28)
    at DOMWorld._waitForSelectorOrXPath (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:478:22)
    at DOMWorld.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:432:17)
    at Frame.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/FrameManager.js:627:47)
    at Frame.<anonymous> (/root/bbb-recorder/node_modules/puppeteer/lib/helper.js:112:23)
    at Page.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/Page.js:1122:29)
    at main (/root/bbb-recorder/liveRTMP.js:73:14)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7) name: 'TimeoutError' }
```

Can you help me, please?

**Comment by jibon57** [comment]
Are you sure the joining link is correct? Check from browser to see if you are able to login using the link. Check BBB API for more information.

**Comment by awe007** [comment]
Hmm...
`When i try this link https://domain.online/bigbluebutton/api/`
i see this:
```
<response>
<returncode>SUCCESS</returncode>
<version>2.0</version>
</response>
```
But when i try full link
`https://domain.online/bigbluebutton/api/join?meetingId=adm-96w-m2a`
i just redirected to main page https://domain.online/b

In Greenlight interface i see this:
```
Name: test1
Created: 2020-05-02 09:37:54 UTC 
Owner: Administrator 
ID: adm-96w-m2a 
Status: Running
```
May be ID is wrong?

**Comment by jibon57** [comment]
The ID is correct. Now you can use BBB API to create joining link because Greenlight don't allow to provide link manually. You can get details from here: http://docs.bigbluebutton.org/dev/api.html#join

**Comment by samuelpetrosyan** [comment]
I have the same problem!!
@jibon57 Can you please describe how can you use live recording or streaming if using Greenlight??

Thanks in advance!

**Comment by awe007** [comment]
Well. I finally got a working link to join the conference. Please tell me how do I get a live broadcast now? Where can I watch it?

**Comment by awe007** [comment]
At the first terminal screen (where **node ffmpegServer.js**) i see this:

```
DATA <Buffer 1a 45 df a3 a3 42 86 81 01 42 f7 81 01 42 f2 81 04 42 f3 81 08 42 82 88 6d 61 74 72 6f 73 6b 61 42 87 81 04 42 85 81 02 18 53 80 67 01 ff ff ff ff ff ... >
DATA <Buffer 42 eb 82 00 3c 80 ff 03 60 7b 4c 42 85 c8 ae 28 06 7e 8d 69 ab 3f 01 de e1 af e8 e6 44 70 9e db 5b fe 27 34 45 8d 33 e6 6d 13 ef ff 5e 5d 5d 8a 9d ed ... >
FFmpeg STDERR: [matroska,webm @ 0x560b56a49980] decoding for stream 0 failed

FFmpeg STDERR: Input #0, matroska,webm, from 'pipe:':
  Metadata:

FFmpeg STDERR:     encoder         : Chrome
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0(eng): Video: h264 (Constrained Baseline), yuv420p(progressive), 1280x720, SAR 1:1 DAR 16:9, 1k tbr, 1k tbn, 2k tbc (default)
    Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)

FFmpeg STDERR: rtmp://tochkavstrechi.online/live2/MyKey: Input/output error

FFmpeg STDIN Error { Error: write EPIPE
    at WriteWrap.afterWrite [as oncomplete] (net.js:868:14) errno: 'EPIPE', code: 'EPIPE', syscall: 'write' }
FFmpeg child process closed, code 1, signal null
connection
FFmpeg STDERR: ffmpeg version 4.2.2-1bbb1~ubuntu16.04 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.12) 20160609
```

**Comment by samuelpetrosyan** [comment]
As I know you should set your RTMP parameters in /config.json.

**Comment by samuelpetrosyan** [comment]
> Well. I finally got a working link to join the conference. Please tell me how do I get a live broadcast now? Where can I watch it?

@awe007, Please, tell how can you get link to join the conference? I still can't.

**Comment by awe007** [comment]
There is my config.json

```
{
	"rtmpUrl": "rtmp://domain.online/live2/MyKey",
	"ffmpegServer": "ws://localhost",
	"ffmpegServerPort": 4000,
	"auth": "mZFZN4yc"
}
```

What i need to do ? Maybe you can give me an example of right settings?

**Comment by awe007** [comment]
For link i use this function

```
_getChecksum(endpoint, query) {
    const secret = '<secret>';
    return shajs('sha1').update(`${endpoint}${query}${secret}`).digest('hex');
  }
```
Endpoint = Join
Query = meetingID=test01&password=mp&fullName=John
Secret = Secret from .env

This function return us checksum for link, and full url to join will be
`http://yourserver.com/bigbluebutton/api/join?meetingID=test01&password=mp&fullName=John&checksum=1234`

**Comment by samuelpetrosyan** [comment]
> There is my config.json
> 
> ```
> {
> 	"rtmpUrl": "rtmp://domain.online/live2/MyKey",
> 	"ffmpegServer": "ws://localhost",
> 	"ffmpegServerPort": 4000,
> 	"auth": "mZFZN4yc"
> }
> ```
> 
> What i need to do ? Maybe you can give me an example of right settings?

If you want to stream to Youtube. You should fill the fields with info provided from Youtube Studio Streaming page.

**Comment by awe007** [comment]
With bbb-recorder can i stream my BBB-Conference to my own another site, like WordPress ?

**Comment by samuelpetrosyan** [comment]
> For link i use this function
> 
> ```
> _getChecksum(endpoint, query) {
>     const secret = '<secret>';
>     return shajs('sha1').update(`${endpoint}${query}${secret}`).digest('hex');
>   }
> ```
> 
> Endpoint = Join
> Query = meetingID=test01&password=mp&fullName=John
> Secret = Secret from .env
> 
> This function return us checksum for link, and full url to join will be
> `http://yourserver.com/bigbluebutton/api/join?meetingID=test01&password=mp&fullName=John&checksum=1234`

How do you use this function??

**Comment by jibon57** [comment]
try: https://github.com/jibon57/bbb-recorder/commit/7959d2bb95eb12a9c4bad62e3443dbec7149319c
